### ISSUE TITLE
Be able to see what --fix does

### ISSUE THREAD

**Issue by ericandrewlewis** [eslint/eslint#8221]
I'd like to be able to see what --fix does, rather than just looking at a diff of the affects after I run it.

**Comment by ericandrewlewis** [comment]
For my purposes, I added a `debug()` line inside the loop that applies each fix [here](https://github.com/eslint/eslint/blob/ca1694badce2ed8688766ebc8b29edf008aef51c/lib/util/source-code-fixer.js#L93), a la

```js
for (const problem of fixes.sort(compareMessagesByFixRange)) {
  debug("Applying fix for problem: " problem.message);
  // ...
}
```

So then to see all the fixes along the way I can run

```bash
DEBUG=eslint:text-fixer,eslint:cli-engine eslint --fix file.js
```

**Comment by kaicataldo** [comment]
@ericandrewlewis What kind of API are you envisioning?

**Comment by ericandrewlewis** [comment]
Knowing what I know now about how ESLint works, I think adding a debug call like the one I [patched into my local version of ESLint](https://github.com/eslint/eslint/issues/8221#issuecomment-285355758) would be useful.

I guess something like `--dry-run` could be useful, but I'm not as interested in that myself since fixing code is a reversable action with source control (`git checkout *`)

**Comment by vitorbal** [comment]
I agree there would be some value in being able to see these for debugging purposes, but that's already the case when you use the --debug flag. You get something like this:

```bash
  eslint:text-fixer Applying fixes +1ms
  eslint:text-fixer Found fixes to apply +0ms
```

Maybe eslint could output a statement like "Fixes were applied." when running out of debug mode, but I think that might be a breaking change.

**Comment by gyandeeps** [comment]
I am not sure whats being asked here. We want to know fix stuff for debugging purposes as an eslint developer or how/what fixes eslint are fixing as a user of eslint?

For debug purpose: We already have a debug call in it.
For eslint consumers: Not sure what is expected but if you run eslint without fix flag vs with fix flag you can see the difference. And now we even tell you how many issues can be fixed using fix flag.

With that i think we are providing enough info to our users.

**Comment by tiriana** [comment]
@gyandeeps I don't know about the autor of this issue, and what was the initial need. But from my perspective, I would like to be able to see that 

`line <line number>, file <file path>, applying fix due to rule <rule name>`

**Comment by platinumazure** [comment]
I think this can be done using `--fix-dry-run` and examining the output in (e.g.) the JSON formatter.
